[USING] rework password finder (#742)
zkSNACKs/WasabiDoc: docs/using-wasabi/PasswordFinder.md

## Limitations

> Wasabi Wallet protects the encrypted secret key with with a user specified password ([BIP 38](/using-wasabi/BIPs.md#bip-38-password-protected-private-key)).

Reviewer: ```suggestion
Wasabi Wallet protects the encrypted secret key with a user specified password ([BIP 38](/using-wasabi/BIPs.md#bip-38-password-protected-private-key)).
```

> One of the features of this method is that it has brute force protection, meaning even if the password is wrong, it still generates a valid (but unused) private key.

Reviewer 2: There is a deep misunderstanding of bip38 here. Bip38 is simply a method to encrypt a private key and encode it as a printable string. This method involves the usage of scrypt to make the brute force attacks harder. 

Any attempt to decrypt the encrypted private key with the wrong password results in an error. It does never return an incorrect private key.

Bip39 is the method to generate a seed from human-readable words. It allows to add additional entropy by using extra words, what we usually refer as the 13th or 25th word. This means that any change any of the words will generate a completely different seed. For example, imagine we don't provide a 13th word but we make a mistake writing the 12 words in our piece of paper, it can be a typo, a mistake in the order of the words or whatever, that will result in a different seed.

In wasabi both bips are in place and they both share something in common: the 13th word used in bip39 is the password used in the encryption (bip38). But brute force protection (bip38) and seed derivation (bip39) are really two very different things.

> One of the features of this method is that it has brute force protection, meaning even if the password is wrong, it still generates a valid [but unused] private key.

Reviewer 3: ```suggestion
One of the features of this method is that it has brute force protection, meaning even if the password is wrong, it still generates a valid (but unused) private key.
```

> One of the features of this method is that it has brute force protection, meaning even if the password is wrong, it still generates a valid (but unused) private key.
> However, the Wasabi Password Finder has the wallet file, and thus it knows the used public keys of the wallet, and it can check the generated private key against the known to be valid public key.

Author: ```suggestion
The Wasabi Password finder requires the wallet file.
```

@lontivero how about this?

Reviewer: How about we simply keep that part as it was.

Author: Because I think this is not accurate... If it is computationally infeasible depends on the password chosen, not on the BIP38 standard per se...
` and for that reason it is computationally infeasible to brute force the password using all the possible combinations.`

Reviewer: we can simply add `(assuming a secure password was chosen)`.
No need to completely re-write all that.

> Wasabi Wallet protects the encrypted secret key with a user specified password ([BIP 38](/using-wasabi/BIPs.md#bip-38-password-protected-private-key)).
> One of the features of this method is that it has brute force protection, meaning even if the password is wrong, it still generates a valid (but unused) private key.
> However, the Wasabi Password Finder has the wallet file, and thus it knows the used public keys of the wallet, and it can check the generated private key against the known to be valid public key.
> It is important to know that Wasabi Password Finder is not for breaking wallet passwords but for finding minor errors (typos) in a roughly known password. 

Reviewer: ```suggestion
Wasabi Wallet protects the encrypted secret key with the same technology used to protect paper wallets ([BIP 38](/using-wasabi/BIPs.md#bip-38-password-protected-private-key)) and for that reason it is computationally infeasible to brute force the password using all the possible combinations. (assuming a secure password was chosen)
It is important to know that Wasabi Password Finder is not for breaking wallet passwords but for finding errors (typos) in an already known password.
```

## Usage

> #### Windows

Reviewer: ```suggestion
### Windows
```

To use Wasabi's command line tools on Windows you have to use `wassabeed.exe` that is inside your `Program Files\WasabiWallet`.

> #### Linus and macOS

Reviewer: ```suggestion
### Linus and macOS
```

Use the `wassabee` command.

```

> $ wassabee help findpassword

Reviewer: Is 'run' not necessary here?

Author: run breaks it.
Tested on Qubes 4 Debian 10 - not sure about Windows though

  -h, --help                 Show Help
```

> For example, there is a called `MagicalCryptoWallet.json`, the correct password is `pass`, but the user thinks it is `pasd`.

Reviewer: ```suggestion
For example, there is a wallet called `MagicalCryptoWallet.json`, the correct password is `pass`, but the user thinks it is `pasd`.
```
